tests/mutex.rs

```diff
@@ -130,64 +123,52 @@ fn lock_uncontested() {
 #[test]
 fn lock_contested() {
     let mutex = Mutex::<u32>::new(0);
-    let mut rt = current_thread::Runtime::new().unwrap();
 
-    rt.block_on(lazy(|| {
-        let mut fut0 = mutex.lock();
-        let guard0 = fut0.poll();    // fut0 immediately gets ownership
-        assert!(guard0.as_ref().unwrap().is_ready());
+    let mut fut0 = spawn(mutex.lock());
+    let guard0 = assert_ready!(fut0.poll()); // fut0 immediately gets ownership
 
-        let mut fut1 = mutex.lock();
-        assert!(!fut1.poll().unwrap().is_ready());  // fut1 is blocked
+    let mut fut1 = spawn(mutex.lock());
+    assert_pending!(fut1.poll());            // fut1 is blocked
 
-        drop(guard0);               // Ownership transfers to fut1
-        let guard1 = fut1.poll();
-        assert!(guard1.as_ref().unwrap().is_ready());
-        future::ok::<(), ()>(())
-    })).unwrap();
+    drop(guard0);                            // Ownership transfers to fut1
+    assert!(fut1.is_woken());
+    assert_ready!(fut1.poll());
 }
 
-// A single Mutex is contested by tasks in multiple threads
-#[test]
-fn lock_multithreaded() {
+//// A single Mutex is contested by tasks in multiple threads
+#[tokio::test]
+async fn lock_multithreaded() {
```

Comment:
shouldn't you have to spawn to be multi threaded? see https://github.com/tokio-rs/tokio/blob/b1c46c4be07ee67305a2fcfc4c3c8ece6d081991/tokio/tests/sync_rwlock.rs#L68

Reply:
Good catch.